runtime/queries/metricsview_aggregation.go

```diff
@@ -76,14 +91,138 @@ func (q *MetricsViewAggregation) Resolve(ctx context.Context, rt *runtime.Runtim
 		return fmt.Errorf("metrics view '%s' does not have a time dimension", q.MetricsView)
 	}
 
-	// Build query
-	sql, args, err := q.buildMetricsAggregationSQL(q.MetricsView, olap.Dialect(), q.ResolvedMVSecurity)
+	// execute druid query
+	sqlString, args, err := q.buildMetricsAggregationSQL(q.MetricsView, olap.Dialect(), q.ResolvedMVSecurity)
 	if err != nil {
 		return fmt.Errorf("error building query: %w", err)
 	}
 
-	// Execute
-	schema, data, err := olapQuery(ctx, olap, priority, sql, args)
+	schema, data, err := olapQuery(ctx, olap, priority, sqlString, args)
+	if err != nil {
+		return err
+	}
+
+	if len(q.PivotOn) == 0 {
+		q.Result = &runtimev1.MetricsViewAggregationResponse{
+			Schema: schema,
+			Data:   data,
+		}
+		return nil
+	}
+
+	var connErr error
+	var db *databasesql.DB
+	dbOnce.Do(func() {
+		connector, connErr = duckdb.NewConnector("", func(conn driver.ExecerContext) error {
+			_, err := conn.ExecContext(context.Background(), "INSTALL 'json'; LOAD 'json';", nil)
+			return err
+		})
+
+		db = databasesql.OpenDB(connector)
+		db.SetMaxOpenConns(1)
+	})
+	if connErr != nil {
+		return connErr
+	}
+
+	conn, err := connector.Connect(ctx)
+	if err != nil {
+		return err
+	}
+	defer conn.Close()
```

Comment:
The `db` variable will be `nil` on every invocation after the first (which triggers `dbOnce`). Generally, this code mixes up `db` and `conn` multiple places.

I would propose moving the entire olapQuery -> temporary table in a temporary DuckDB handle logic into a util function. Something like:
```
schema, data, err := olapQuery(...)
if err != nil {...}

conn, tableName, release, err := ephemeralDuckDBForResult(schema, data)
// ...
```

If you use that for Druid, combined with a call to `WithConnection` for the DuckDB case (which provides access to the underlying `conn *sql.Conn` in the callback), you might be able to have common logic for doing the pivot (and export) for both. But it's also worth considering treating both cases completely separately, even if it's more verbose.

Regardless, I think the various bugs around connection management here show that it's worth putting some time into thinking about how to make the logic flows here as simple to understand/review as possible.